cli/src/pcluster/templates/cdk_builder_utils.py

```diff
@@ -877,6 +907,86 @@ def _build_policy(self) -> List[iam.PolicyStatement]:
         ]
 
 
+class LoginNodesIamResources(NodeIamResourcesBase):
+    """Construct defining IAM resources for a login node."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        config: BaseClusterConfig,
+        node: LoginNodesPool,
+        shared_storage_infos: dict,
+        name: str,
+    ):
+        super().__init__(scope, id, config, node, shared_storage_infos, name)
+
+    def _build_policy(self) -> List[iam.PolicyStatement]:
+        return [
+            iam.PolicyStatement(
+                sid="Ec2",
+                actions=[
+                    "ec2:DescribeInstances",
+                    "ec2:DescribeInstanceStatus",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=["*"],
+            ),
+            iam.PolicyStatement(
+                sid="S3",
+                actions=[
+                    "s3:GetObject",
+                    "s3:PutObject",
+                    "s3:DeleteObject",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=[
+                    self._format_arn(
+                        service="s3",
+                        resource="{0}-aws-parallelcluster/*".format(Stack.of(self).region),
+                        region="",
+                        account="",
+                    )
+                ],
+            ),
+            iam.PolicyStatement(
+                sid="DynamoDB",
+                actions=[
+                    "dynamodb:GetItem",
+                    "dynamodb:PutItem",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=["*"],
+            ),
+            iam.PolicyStatement(
+                sid="Batch",
+                actions=[
+                    "batch:SubmitJob",
+                    "batch:DescribeJobs",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=["*"],
+            ),
+            iam.PolicyStatement(
+                sid="IAM",
+                actions=[
+                    "iam:AssumeRole",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=["*"],
+            ),
+            iam.PolicyStatement(
+                sid="CloudWatchLogs",
+                actions=[
+                    "logs:GetLogEvents",
+                    "logs:FilterLogEvents",
+                ],
+                effect=iam.Effect.ALLOW,
+                resources=["*"],
```

Comment:
I don't think the login nodes need all these policies for now.

Reply:
okay, I will delete _build_policy function for now.

Reply:
I delete most of them, but keep EC2 Policy, cause _build_policy is a abstract method, must have. And if I return an empty list, two unit test will fail. So for the convenience of adding later, I keep one for now.
```
>           raise RuntimeError(resp.error) from JavaScriptError(resp.stack)
E           RuntimeError: Error: Resolution error: Supplied properties not correct for "CfnPolicyProps"
E             policyDocument: required but missing.
```

Reply:
I suggest checking the 2 failing unit tests to understand how they can be improved.
Adding policies as placeholders is a bit risky since they can be forgotten unintentionally.

Reply:
I found that if I leave this function like this the unit test could all pass.
```
def _build_policy(self) -> List[iam.PolicyStatement]:
        return [
            iam.PolicyStatement(),
        ]
```

Reply:
I removed all the LoginNodesIamResource related code for now.